crates/store/re_rrdp_comms/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+}
+
+// ----------------------------------------------------------------------------
+
+/// Stream an rrd file from an RRDP server.
+///
+/// `on_msg` can be used to wake up the UI thread on Wasm.
+pub fn stream_recording(
+    url: String,
+    on_msg: Option<Box<dyn Fn() + Send + Sync>>,
+) -> Result<re_smart_channel::Receiver<LogMsg>, InvalidAddressError> {
+    re_log::debug!("Loading {url}…");
+
+    let address = Address::from_str(&url)?;
+
+    let (tx, rx) = re_smart_channel::smart_channel(
+        re_smart_channel::SmartMessageSource::RrdpStream { url: url.clone() },
+        re_smart_channel::SmartChannelSource::RrdpStream { url: url.clone() },
+    );
+
+    spawn_future(async move {
+        if let Err(err) = stream_recording_async(tx, address, on_msg).await {
+            re_log::warn!(
+                "Error while streaming {url}: {}",
+                re_error::format_ref(&err)
+            );
+        }
+    });
+
+    Ok(rx)
+}
+
+#[cfg(target_arch = "wasm32")]
+fn spawn_future<F>(future: F)
+where
+    F: std::future::Future<Output = ()> + 'static,
+{
+    wasm_bindgen_futures::spawn_local(future);
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn spawn_future<F>(future: F)
+where
+    F: std::future::Future<Output = ()> + 'static + Send,
+{
+    tokio::spawn(future);
+}
+
+async fn stream_recording_async(
+    tx: re_smart_channel::Sender<LogMsg>,
+    address: Address,
+    on_msg: Option<Box<dyn Fn() + Send + Sync>>,
+) -> Result<(), StreamError> {
+    use tokio_stream::StreamExt as _;
+
+    let Address {
+        addr_port,
+        recording_id,
+    } = address;
+
+    if addr_port.starts_with("0.0.0.0:") {
+        re_log::warn!("Attempting to connect to IP 0.0.0.0. This will often fail. You likely you want connect to 127.0.0.1 instead.");
+    }
+
+    let http_addr = format!("http://{addr_port}");
+    re_log::debug!("Connecting to {http_addr}…");
+
+    let mut client = {
+        #[cfg(target_arch = "wasm32")]
+        let tonic_client = tonic_web_wasm_client::Client::new_with_options(
+            http_addr,
+            tonic_web_wasm_client::options::FetchOptions::new()
+                .mode(tonic_web_wasm_client::options::Mode::Cors), // I'm not 100% sure this is needed, but it felt right.
+        );
+
+        #[cfg(not(target_arch = "wasm32"))]
+        let tonic_client = tonic::transport::Endpoint::new(http_addr)?
+            .connect()
+            .await?;
+
+        StorageNodeClient::new(tonic_client)
+    };
+
+    client = client.max_decoding_message_size(1024 * 1024 * 1024);
+
+    re_log::debug!("Fetching {recording_id}…");
+
+    let mut resp = client
+        .fetch_recording(FetchRecordingRequest {
+            recording_id: Some(RecordingId {
+                id: recording_id.clone(),
+            }),
+        })
+        .await
+        .map_err(TonicStatusError)?
+        .into_inner();
+
+    drop(client);
+
+    // TODO(jleibs): Does this come from RDP?
```

Comment:
(I guess this could be part of the protocol where first message for ``FetchRecording`` endpoint is just the ``StoreId``. But below certainly works as well)